src/client/websocket/WebSocketShard.js

```diff
@@ -324,7 +310,7 @@ class WebSocketShard extends EventEmitter {
    * @private
    */
   onClose(event) {
-    this.closeSequence = this.sequence;
+    if (this.sequence && this.sequence !== -1) this.closeSequence = this.sequence;
```

Comment:
`this.sequence` can only be falsy when it is `0` (considering all assignments are to numbers), and it is redudant with `this.sequence !== -1`, since it cannot be falsy and `-1`:
```suggestion
    if (this.sequence !== -1) this.closeSequence = this.sequence;
```